tests/bsim/bluetooth/host/privacy/device/src/test_undirected_peripheral.c

```diff
@@ -421,26 +421,36 @@ static void test_peripheral_main(void)
 
 	update_adv_params(adv, CONN_NSCAN, RPA);
 	start_adv(adv);
+
 	WAIT_FOR_FLAG(paired_flag);
+
 	disconnect();
 	stop_adv(adv);
 
 	update_adv_params(adv, adv_param, test_addr_type);
 	start_adv(adv);
+
 	/* (the connection with identity should fail with privacy network mode) */
 	if (connectable_test) {
 		wait_for_connection();
 		disconnect();
 	} else if (scannable_test && use_ext_adv) {
 		wait_for_scanned();
 	}
-	stop_adv(adv);
+
+
+	/* The zephyr link layer doesn't behave as expected, it only pends one
+	 * report instead of the two mandated by the specification when
+	 * performing active scanning. Remove workaround when #56795 is
+	 * resolved.
+	 */
+	if (!IS_ENABLED(CONFIG_BT_LL_SW_SPLIT)) {
+		stop_adv(adv);
+	}
```

Comment:
```suggestion


	/* The zephyr link layer immediately stops any radio transmission on advertising stop,
	 * hence add a delay before disabling advertising to ensure the scan response PDU
	 * transmission is completed on-air.
	 */
	 k_sleep(K_MSEC(10));
	stop_adv(adv);
```

Reply:
Can we remove the stop instead of adding a sleep, and just leave a comment?

```suggestion


	/* The HCI LE Scan Request Received event does not guarantee that
	 * the advertiser has already sent the response to the request. We leave
	 * the advertiser running in case the response is yet to be sent.
	 */
```

Reply:
haha I just saw the comment right after I pushed. did what you wanted anyways :)